src/rand.rs

```diff
@@ -51,19 +51,14 @@ pub fn xor_pseudo_rng_u32() -> u32 {
 // because they will be in the range [a-zA-Z0-9]. The return vector could be any
 // size between 0 and 4.
 fn xor_pseudo_rng_u8_alphanumerics(rand_fn: &dyn Fn() -> u32) -> Vec<u8> {
-    let mut r = vec![];
-
-    fn between(lower: u8, upper: u8, to_check: u8) -> bool {
-        (to_check >= lower) && (to_check <= upper)
-    }
-
-    for n in &rand_fn().to_ne_bytes() {
-        // Upper / Lower alphabetics and numbers.
-        if between(48, 57, *n) || between(65, 90, *n) || between(97, 122, *n) {
-            r.push(*n);
-        }
-    }
-    r
+    rand_fn()
+        .to_ne_bytes()
+        .to_vec()
+        .drain(..)
```

Comment:
nit: I find `drain` a bit confusing here, any reason for not using `into_iter()`?

Reply:
`into_iter` consumes the vector as well, so from that point of view I found drain to be more clear about the operation being performed. I don't have a strong preference though.